pkg/netstore/netstore.go

```diff
@@ -65,7 +82,19 @@ func (s *store) Put(ctx context.Context, mode storage.ModePut, chs ...swarm.Chun
 			return nil, storage.ErrInvalidChunk
 		}
 	}
-	return s.Storer.Put(ctx, mode, chs...)
+	exist, err = s.Storer.Put(ctx, mode, chs...)
+	if err != nil {
+		return nil, err
+	}
+	// if callback is defined, call it for every new, valid chunk
+	if s.deliveryCallback != nil {
+		for i, exists := range exist {
+			if !exists {
+				go s.deliveryCallback(chs[i])
```

Comment:
i am not 100% sure that a closure is not needed here. cc @janos

Reply:
also, having given this further thought, i am not sure that we want to have this here in general. 

why: this means that every incoming chunk we check if its a trojan chunk. for example, this means that every chunk we retrieve with the `retrieval` protocol will be checked to be a trojan chunk. this is unnecessary computation that need not be done.

the way i see it, trojan chunks have two ingress points - one on pushsync protocol, the second on pullsync (for mailboxing). not sure if this needs to be on both (it might be that pushsync is only necessary right now), but it would be more efficient to have this hook only called in push and pull sync protocols.